DS/DS4/GuestHouse.java

```diff
@@ -0,0 +1,2575 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Scanner;
+class RoomAlotement {
+	public List<String> showRoomStatus(Enumeration<Integer> roomNumbers,Hashtable<Integer, String> rooms,int roomNumber) {
+		List< String> roomStatus=new ArrayList<String>();
+		roomNumbers = rooms.keys();	// room numbers hold all the keys (room numbers) from hash table
+		while(roomNumbers.hasMoreElements()) {
+	         roomNumber = roomNumbers.nextElement();
+	         roomStatus.add("Room "+(roomNumber) + " : " +rooms.get(roomNumber));
+	           // show the status of each room
+		}
+		return roomStatus;
+	}
+	public String allotRoom(Hashtable<Integer, String>rooms,int age,String name,int numberOfRooms) {
```

Comment:
Why is the 'rooms' variable has to be passed to each and every method all the time, can't it be a class member?